script.globalsearch/lib/gui.py

```diff
@@ -83,20 +86,20 @@ def _init_items(self):
         self.oldfocus = 0
 
     def _set_view(self):
-        try:
-            vid = ADDON.getSettingInt('view')
-        except:
-            vid = 0
-        if vid:
-            xbmc.executebuiltin('Container.SetViewMode(%i)' % vid)
-        else:
-            # no view will be loaded unless we call SetViewMode, might be a bug...
-            xbmc.executebuiltin('Container.SetViewMode(-1)')
+        # no view will be loaded unless we call SetViewMode, might be a bug...
+        xbmc.executebuiltin('Container.SetViewMode(0)')
+        vid = ADDON.getSettingInt('view')
+        # kodi bug: need to call Container.SetViewMode twice
```

Comment:
Probably a good idea to open an issue at the core repo explaining this? It's a problem add-ons implement hacks to overcome Kodi flaws that are otherwise never known (nor will ever have a fix if they are not public)

Reply:
Setviewmodes is also normally blocked for add-ons, but I trust you (as the skin authority) know what you are doing here

Reply:
> Probably a good idea to open an issue at the core repo explaining this?
there's a thread on the forum were the issue was discussed (kodi dev involved as well):
https://forum.kodi.tv/showthread.php?tid=329208
once i figured out what broke it, i've mentioned on the PR as well:
https://github.com/xbmc/xbmc/pull/13542#issuecomment-370537321

> Setviewmodes is also normally blocked for add-ons
yup, that certainly applies to plugins changing the viewmode of a skin.
here, i'm just changing the viewmode inside my own little globalsearch window.
it has no effect outside of this particular script window.